packages/repository/src/legacy-juggler-bridge.ts

```diff
@@ -43,7 +52,7 @@ export function bindModel<T extends typeof juggler.ModelBase>(
  * @param p Promise or void
  */
 /* tslint:disable-next-line:no-any */
-function ensurePromise(p: juggler.PromiseOrVoid<any>): Promise<any> {
+function ensurePromise<T>(p: juggler.PromiseOrVoid<T>): Promise<T> {
```

Comment:
**!!!THIS!!!**

Before my change, `ensurePromise` was silently converting anything into anything else as far as the compiler typechecks went. For example, it happily converted `Promise<PersistedModel>` to `Promise<Entity>`, even though these classes were unrelated.

When I fixed the signature to preserve the type of the promise, suddenly a major design problem in `DefautCrudRepository` was discovered by the compiler - methods say they will return Entity instances, but in fact they were returning PersistedModel instances.

A valuable lesson for all of us - `any` is evil and must be used very carefully.

Reply:
+1: